tests/phpunit/tests/post/getPageByTitle.php

```diff
@@ -0,0 +1,242 @@
+				'post_title' => 'foo',
+			)
+		);
+
+		$found = get_page_by_title( 'foo' );
+
+		$this->assertSame( $page, $found->ID );
+	}
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_inherit() {
+		$page = self::factory()->post->create(
+			array(
+				'post_type'   => 'page',
+				'post_title'  => 'foo',
+				'post_status' => 'inherit',
+			)
+		);
+
+		$found = get_page_by_title( 'foo' );
+
+		$this->assertSame( $page, $found->ID );
+	}
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_should_obey_post_type() {
+		register_post_type( 'wptests_pt' );
+
+		$page = self::factory()->post->create(
+			array(
+				'post_type'  => 'wptests_pt',
+				'post_title' => 'foo',
+			)
+		);
+
+		$found = get_page_by_title( 'foo' );
+		$this->assertNull( $found );
+
+		$found = get_page_by_title( 'foo', OBJECT, 'wptests_pt' );
+		$this->assertSame( $page, $found->ID );
+	}
+
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_should_hit_cache() {
+		global $wpdb;
+
+		$page = self::factory()->post->create(
+			array(
+				'post_type'  => 'page',
+				'post_title' => 'foo',
+			)
+		);
+
+		// Prime cache.
+		$found = get_page_by_title( 'foo' );
+		$this->assertSame( $page, $found->ID );
+
+		$num_queries = $wpdb->num_queries;
+
+		$found = get_page_by_title( 'foo' );
+		$this->assertSame( $page, $found->ID );
+		$this->assertSame( $num_queries, $wpdb->num_queries );
+	}
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_bad_path_should_be_cached() {
+		global $wpdb;
+
+		// Prime cache.
+		$found = get_page_by_title( 'foo' );
+		$this->assertNull( $found );
+
+		$num_queries = $wpdb->num_queries;
+
+		$found = get_page_by_title( 'foo' );
+		$this->assertNull( $found );
+		$this->assertSame( $num_queries, $wpdb->num_queries );
+	}
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_bad_path_served_from_cache_should_not_fall_back_on_current_post() {
+		global $wpdb, $post;
+
+		// Fake the global.
+		$post = self::factory()->post->create_and_get();
+
+		// Prime cache.
+		$found = get_page_by_title( 'foo' );
+		$this->assertNull( $found );
+
+		$num_queries = $wpdb->num_queries;
+
+		$found = get_page_by_title( 'foo' );
+		$this->assertNull( $found );
+		$this->assertSame( $num_queries, $wpdb->num_queries );
+
+		unset( $post );
+	}
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_cache_should_not_match_post_in_different_post_type_with_same_path() {
+		global $wpdb;
+
+		register_post_type( 'wptests_pt' );
+
+		$p1 = self::factory()->post->create(
+			array(
+				'post_type'  => 'page',
+				'post_title' => 'foo',
+			)
+		);
+
+		$p2 = self::factory()->post->create(
+			array(
+				'post_type'  => 'wptests_pt',
+				'post_title' => 'foo',
+			)
+		);
+
+		// Prime cache for the page.
+		$found = get_page_by_title( 'foo' );
+		$this->assertSame( $p1, $found->ID );
+
+		$num_queries = $wpdb->num_queries;
+
+		$found = get_page_by_title( 'foo', OBJECT, 'wptests_pt' );
+		$this->assertSame( $p2, $found->ID );
+		$num_queries ++;
```

Comment:
I don't think a WordPress coding standard is defined here but `$thing++` is far more common in the code base than `$thing ++`.

Feel free to ignore this whilst muttering "Cool story, Peter".

Reply:
Fixed.